src/twisted/web/test/test_http.py

```diff
@@ -2986,6 +2986,17 @@ def test_ne(self):
         self.assertNotEqual(req, http.Request(DummyChannel(), False))
 
 
+    def test_hashable(self):
+        """
+        A L{http.Request} is hashable.
+        """
+        req = http.Request(DummyChannel(), False)
+        try:
+            hash(req)
+        except TypeError as e:
+            self.fail("http.Request must be hashable: {}".format(e))
```

Comment:
I think it's OK to drop the enclosing `try` `except`.  Other tests that call functions that raise an exception on failure, such as those that check interface compliance with `zope.interface.verify.verifyObject`, don't use `fail` to produce a nicer error message, and eliminating this would boost coverage.